libflux/go/libflux/parser.go

```diff
@@ -63,3 +76,14 @@ func Parse(s string) *ASTFile {
 	runtime.SetFinalizer(f, free)
 	return f
 }
+
+func ParseIntoFbs(s string) *ast.Package {
+	cstr := C.CString(s)
+	defer C.free(unsafe.Pointer(cstr))
+
+	ptr := C.flux_parse_fb(cstr)
+	f := &AstBuf{ptr: ptr}
+	runtime.SetFinalizer(f, free)
```

Comment:
Do you need `SetFinalizer` here?  You can just use `defer` instead, right?  In the `Parse` method above, we're returning an `*ASTFile` that contains a pointer that needs to be freed.  But here you're returning an AST that lives entirely in memory managed by Go.

Reply:
Yes, you are right.